app/models/service_request.rb

```diff
@@ -598,6 +598,16 @@ def has_non_first_draft_ssrs?
     sub_service_requests.where.not(status: 'first_draft').any?
   end
 
+  def ssrs_associated_with_service_provider (service_provider)
```

Comment:
I'll make a note to look at this later!  I definitely have some refactoring to do once all the different email scenarios are done.  Thanks for the input!